packages/server/lib/gui/auth.ts

```diff
@@ -213,11 +172,61 @@ const _launchNativeAuth = Promise.method((loginUrl, sendMessage) => {
   })
 })
 
+/**
+ * Grouping internal APIs under separate export to allow for stubbing
+ * in public API tests.
+ */
+const _internal = {
+  buildLoginRedirectUrl,
+  buildFullLoginUrl,
+  getOriginFromUrl,
+  launchServer,
+  stopServer,
+  launchNativeAuth,
+}
+
+/**
+ * @returns a promise that is resolved with a user when auth is complete or rejected when it fails
+ */
+const start = (onMessage, utmCode) => {
+  function sendMessage (type, name, arg1) {
+    onMessage({
+      type,
+      name,
+      message: errors.getMsgByType(name, arg1),
+      browserOpened: authRedirectReached,
+    })
+  }
+
+  authRedirectReached = false
+
+  return user.getBaseLoginUrl()
+  .then((baseLoginUrl) => {
+    return _internal.launchServer(baseLoginUrl, sendMessage, utmCode)
+  })
+  .then(() => {
+    return _internal.buildLoginRedirectUrl(server)
+  })
+  .then((loginRedirectUrl) => {
+    debug('Trying to open native auth to URL ', loginRedirectUrl)
```

Comment:
```suggestion
    debug('Trying to open native auth to URL', loginRedirectUrl)
```
```suggestion
    debug('Trying to open native auth to URL ', loginRedirectUrl)
```

Reply:
Could also do this I believe:

```suggestion
    debug('Trying to open native auth to URL %s', loginRedirectUrl)
```